tools/api-markdown-documenter/src/test/SnapshotTestUtilities.ts

```diff
@@ -3,11 +3,194 @@
  * Licensed under the MIT License.
  */
 
-import { FileSystem, NewlineKind } from "@rushstack/node-core-library";
+import Path from "node:path";
+
+import type { ApiModel } from "@microsoft/api-extractor-model";
+import { FileSystem } from "@rushstack/node-core-library";
 import { expect } from "chai";
 import { compare } from "dir-compare";
 
-import { type FileSystemConfiguration } from "../FileSystemConfiguration.js";
+import {
+	transformApiModel,
+	type ApiItemTransformationConfiguration,
+} from "../api-item-transforms/index.js";
+import type { Suite } from "mocha";
+import { loadModel } from "../LoadModel.js";
+import type { DocumentNode } from "../documentation-domain/index.js";
+
+/**
+ * End-to-end snapshot test configuration.
+ *
+ * @remarks Generates a test suite with a test for each combination of API Model and test configuration.
+ */
+export interface EndToEndSuiteConfig<TRenderConfig> {
+	/**
+	 * Name of the outer test suite.
+	 */
+	readonly suiteName: string;
+
+	/**
+	 * Path to the directory where all suite test output will be written for comparison against checked-in snapshots.
+	 *
+	 * @remarks
+	 * Individual tests' output will be written to `<temporaryOutputDirectoryPath>/<{@link ApiModelTestOptions.modelName}>/<{@link ApiItemTransformationTestOptions.configName}>/<{@link RenderTestOptions.configName}>`.
+	 */
+	readonly temporaryOutputDirectoryPath: string;
+
+	/**
+	 * Path to the directory containing the checked-in snapshots for comparison in this suite.
+	 *
+	 * @remarks
+	 * Individual tests' output will be written to `<temporaryOutputDirectoryPath>/<{@link ApiModelTestOptions.modelName}>/<{@link ApiItemTransformationTestOptions.configName}>/<{@link RenderTestOptions.configName}>`.
+	 */
+	readonly snapshotsDirectoryPath: string;
+
+	/**
+	 * The end-to-end test scenario to run against the API model.
+	 * Writes the output to the specified directory for snapshot comparison.
+	 */
+	render(
+		document: DocumentNode,
+		renderConfig: TRenderConfig,
+		outputDirectoryPath: string,
+	): Promise<void>;
+
+	/**
+	 * The models to test.
+	 */
+	readonly apiModels: readonly ApiModelTestOptions[];
+
+	/**
+	 * Test configurations to run against each API Model.
+	 */
+	readonly testConfigs: readonly EndToEndTestConfig<TRenderConfig>[];
+}
+
+/**
+ * API Model test options for a test.
+ */
+export interface ApiModelTestOptions {
+	/**
+	 * Name of the API Model being tested.
+	 */
+	readonly modelName: string;
+
+	/**
+	 * Path to the directory containing the API Model.
+	 */
+	readonly directoryPath: string;
+}
+
+/**
+ * API Item transformation options for a test.
+ */
+export interface EndToEndTestConfig<TRenderConfig> {
```

Comment:
I think EndToEndTestConfig and endToEndTestSuite would be more discoverable in their own file.

Reply:
(Unless "Snapshot Test" is the same as "End-to-End Test," in which case the names should be unified.)